pkg/test/e2e/default_app_catalog_tests/application_test.go

```diff
@@ -0,0 +1,468 @@
+
+	client, cleanup, tLogger, err := createUserCluster(ctx, t, logger, seedClient)
+	if cleanup != nil {
+		defer cleanup()
+	}
+
+	if err != nil {
+		t.Fatalf("failed to create user cluster: %v", err)
+	}
+
+	testUserCluster(ctx, t, tLogger, client)
+}
+
+//gocyclo:ignore
+func testUserCluster(ctx context.Context, t *testing.T, log *zap.SugaredLogger, client ctrlruntimeclient.Client) {
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: applicationNamespace,
+		},
+	}
+
+	// Create the namespace in Kubernetes
+	err := client.Create(ctx, namespace)
+	if err != nil {
+		t.Fatalf("%v", err)
+	} else {
+		log.Infof("Namespace %s created", namespace.Name)
+	}
+
+	applicationRefName := applicationName
+	if applicationName == "gpu-operator" {
+		applicationRefName = "nvidia-gpu-operator"
+	}
+
+	application := appskubermaticv1.ApplicationInstallation{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      applicationName,
+			Namespace: applicationNamespace,
+		},
+		Spec: appskubermaticv1.ApplicationInstallationSpec{
+			Namespace: &appskubermaticv1.AppNamespaceSpec{
+				Name:   applicationNamespace,
+				Create: true,
+			},
+			ApplicationRef: appskubermaticv1.ApplicationRef{
+				Name:    applicationRefName,
+				Version: applicationVersion,
+			},
+		},
+	}
+
+	if applicationName == "cluster-autoscaler" {
+		valuesBlock := `
+cloudProvider: clusterapi
+clusterAPIMode: incluster-incluster
+autoDiscovery:
+  namespace: kube-system
+image:
+  # 'Cluster.AutoscalerVersion' is injected by KKP based on the Kubernetes version of the cluster.
+  tag: '{{ .Cluster.AutoscalerVersion }}'
+extraEnv:
+  CAPI_GROUP: cluster.k8s.io
+rbac:
+  create: true
+  pspEnabled: false
+  clusterScoped: true
+  serviceAccount:
+    annotations: {}
+    create: true
+    name: "cluster-autoscaler-clusterapi-cluster-autoscaler"
+    automountServiceAccountToken: true
+extraObjects:
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: ClusterRole
+  metadata:
+    name: cluster-autoscaler-management
+  rules:
+  - apiGroups:
+    - cluster.k8s.io
+    resources:
+    - machinedeployments
+    - machinedeployments/scale
+    - machines
+    - machinesets
+    verbs:
+    - get
+    - list
+    - update
+    - watch
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: ClusterRoleBinding
+  metadata:
+    name: cluster-autoscaler-clusterapi-cluser-autoscaler
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: ClusterRole
+    name: cluster-autoscaler-management
+  subjects:
+  - kind: ServiceAccount
+    name: cluster-autoscaler-clusterapi-cluster-autoscaler
+    # 'Release.Namespace' is injected by Helm.
+    namespace: '{{ "{{.Release.Namespace}}" }}'
+`
+		application.Spec.ValuesBlock = valuesBlock
+	}
+
+	log.Infof("Creating an ApplicationInstallation")
+
+	err = client.Create(ctx, &application)
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+
+	time.Sleep(860 * time.Second)
```

Comment:
Is this just here for debugging the job? Should probably be removed since you're doing a `wait.PollLog` below already :)